src/util.cpp

```diff
@@ -459,7 +459,6 @@ bool ArgsManager::GetBoolArg(const std::string& strArg, bool fDefault) const
 
 bool ArgsManager::SoftSetArg(const std::string& strArg, const std::string& strValue)
 {
-    LOCK(cs_args);
```

Comment:
Without this, theoretically there is race to `ForceSetArg` below:
1. two threads get `false` from `IsArgSet("foo")`;
2. one will override the other's value.

Reply:
Ah, good point. Thanks for reviewing!